server/app/auth/oidc/CiviformOidcProfileCreator.java

```diff
@@ -162,17 +165,50 @@ protected CiviFormProfileData mergeCiviFormProfile(
               account -> {
                 accountRepositoryProvider
                     .get()
-                    .updateSerializedIdTokens(account, sessionId.get(), oidcProfile);
+                    .updateSerializedIdTokens(account, sessionId, oidcProfile);
               })
           .join();
     }
 
     return civiformProfile.getProfileData();
   }
 
-  private Optional<String> getSessionId(WebContext context) {
+  private void setSessionIdInProfile(CiviFormProfile profile, WebContext context) {
     PlayWebContext playWebContext = (PlayWebContext) context;
```

Comment:
(question) Is it possible to recover if this somehow wasn't a PlayWebContext? If not probably just leave this, if we can maybe do some type checking and prevent the hard cast from failing.

Reply:
It is not possible to recover.

We already make this cast in a few locations: https://github.com/search?q=repo%3Aciviform%2Fciviform+%22%28PlayWebContext%29%22&type=code